app/services/publishers/ats_api/update_vacancy_service.rb

```diff
@@ -0,0 +1,32 @@
+module Publishers
+  module AtsApi
+    class UpdateVacancyService
+      extend OrganisationFetcher
+
+      class << self
+        def call(vacancy, params)
+          if vacancy.update(sanitised_params(params))
+            { success: true }
+          else
+            { success: false, errors: format_errors(vacancy.errors.messages) }
+          end
+        end
+
+        private
+
+        attr_reader :vacancy, :params
+
+        def sanitised_params(params)
+          organisations = fetch_organisations(params[:schools])
+          raise ActiveRecord::RecordNotFound, "No valid organisations found" if organisations.blank?
```

Comment:
:raising_hand_man: The response behaviour between the create & update when the wrong/no org is provided is inconsistent.

The `create` service raises an `InvalidOrganisationError`, that turns into a `:render_bad_request` response. Should we align both? I would rather have the `create` approach than a `404` when passing a wrong/missing value in a `PUT/PATCH` action. Would leave the `404` exclusively for the wrong vacancy id.

In an ideal scenario, I think this should be added in both `create/update` as a `:unprocessable_entity, errors: "trust_uid/school_urns: No valid organisations found"`, to match the validation error approach.